### PR TITLE
Feature/bab 24 basic usage

### PR DESCRIPTION
**Pull request by baoquandinh**
Added interfaces for league's "All game data" endpoint
Changed prefixes of accepted commands from '-' to '.'

### CODE REVIEW

**Comment by NicholasMata** [comment]
Remove the interfaces for LiveClient API 👎

**Comment by NicholasMata** [comment]
Updated the ticket to reflect what was discussed. please remove the . prefix

**Comment by baoquandinh** [comment]
> Remove the interfaces for LiveClient API 👎

Done

File: src/index.ts [hunk 1]
```diff
@@ -20,8 +20,7 @@ const run = async (args: IArgs) => {
     if (botWasMention && !message.author.bot) {
       let messageContent = message.content.split(' ');
       messageContent.shift()
-      const newMessage = messageContent.join(' ');
-      const service = await Dispatcher(newMessage);
+      const service = await Dispatcher(messageContent);
```
Comment (NicholasMata) [inline src/index.ts]: This does not return a service so the variable should not be called `service` It should be something like newMessage

File: src/services/help_service.ts [hunk 2]
```diff
@@ -7,11 +7,10 @@ export class HelpService implements IService {
...
   }
   buildMessage(): Promise<string> {
-    return Promise.resolve(`That service may not exist, here are the possible services I have:
-        League
-        --------
-        -league
-        -lol
-        -leagueoflegends`);
+    return Promise.resolve(`\`\`\`Usage
+    @BuildABot [command] [arguments]
```
Comment (NicholasMata) [inline src/services/help_service.ts]: `@BuildABot` should be the bots name

**Review by NicholasMata — CHANGES_REQUESTED** [review]

File: src/services/help_service.ts (hunk 2 above)
Comment (NicholasMata) [reply src/services/help_service.ts]: Remove the idents

Comment (NicholasMata) [reply src/services/help_service.ts]: The outputted message should match the ticket in terms for formatted

Comment (baoquandinh) [reply src/services/help_service.ts]: Fixed

File: src/index.ts (hunk 1 above)
Comment (baoquandinh) [reply src/index.ts]: Cleaned up message processing logic

File: src/index.ts [hunk 3]
```diff
@@ -14,16 +14,14 @@ const run = async (args: IArgs) => {
...
-      messageContent.shift()
-      const newMessage = messageContent.join(' ');
-      const service = await Dispatcher(newMessage);
+    const body = message.content;
+    console.log(body)
+    if (body.startsWith(`<@!${botUser.id}>`) && !message.author.bot) {
+      let components = body.split(" ");
+      components[0] = botUser.username
+      const service = await Dispatcher(components);
       if (!service) return;
```
Comment (NicholasMata) [inline src/index.ts]: Rename this variable and remove the null/undefined check as this can never be null or undefined I believe

Comment (baoquandinh) [reply src/index.ts]: Renamed it to "response" and removed the null/undefined check

File: src/tasks/dispatcher.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  if (!botMention) throw Error('Dispatcher requires components to have length greater than 0')
+  let service: IService = new HelpService(botMention);
+  if (components.length === 0) return service.buildMessage();
+
+  const botService = components[0].trim().toLowerCase();
+  console.log(botService);
+  switch (true) {
+    case LeagueService.canProcess(botService):
+      service = new LeagueService();
+      service.helpMessage;
```
Comment (NicholasMata) [inline src/tasks/dispatcher.ts]: Why is this line here?
